src/models/sessionBus.js

```diff
@@ -23,12 +23,15 @@ const {GObject} = imports.gi;
 const {FlatpakSharedModel} = imports.models.shared;
 const {FlatsealOverrideStatus} = imports.models.overrideStatus;
 
+/* https://dbus.freedesktop.org/doc/dbus-specification.html */
+const EXP = /^(([A-Z]|[a-z]|[0-9]|_|-)+)(\.(([A-Z]|[a-z]|[0-9]|_|-)+))+(\.\*){0,1}$/;
```

Comment:
- hyphens are not allowed
- regex can be simplified (`([A-Z]|[a-z])` -> `([A-Za-z])`)
- names must not start with an digit

```suggestion
const EXP = /^([A-Za-z_][A-Za-z0-9_]+)(\.[A-Za-z_][A-Za-z0-9_]+)+(\.\*)?$/;
```

(untested suggestion)

Reply:
This RegEx is not from me. It is already used in Flatseal.
https://github.com/tchx84/Flatseal/blob/562ddec1b12c2a41c64e08daedf7576681f41345/src/widgets/busNameRow.js#L27-L28
It is used to display the valid icon in the D-Bus edit field. If it is wrong, it should be fixed in both places.

Reply:
hypens *are* allowed, but just on the last segment of the name